shared/events.js

```diff
@@ -377,11 +389,10 @@ export class DeleteAttribute extends HashableAction {
 
 export class MailboxContract extends HashableContract {
   static vuex = MailboxContract.Vuex({
-    state: { messages: [], nextId: 0 },
+    state: { messages: [] },
     mutations: {
-      // TODO: verify the correctness of 'nextId' in more detail later
-      PostMessage (state, data) { state.messages.push({...data, id: state.nextId++}) },
-      AuthorizeSender (state, data) { state.authorizations[AuthorizeSender.authorization.name].data = data.sender }
+      PostMessage (state, {data, hash}) { state.messages.push({...data, hash}) },
```

Comment:
By using the splat you are confusing the data contents with the hash, which belong on "separate levels". I.e. what if "data.hash" exists?